lib/1.2/dml-builtins.dml

```diff
@@ -1872,6 +1878,14 @@ template port {
     parameter objtype = "port";
 }
 
+template subdevice {
+    is object;
+    parameter objtype = "subdevice";
+    if ($this.name == "port") {
+         error "A subdevice may not be named 'port'";
```

Comment:
... but may be named 'bank'? How come?

Reply:
In DML 1.2, all objects have a parameter `bank`, and `name` cannot be overridden. So no need for an explicit check!